python/ray/data/tests/test_namespace_expressions.py

```diff
@@ -0,0 +1,591 @@
+"""Tests for list, string, and struct namespace expressions.
+
+This module tests the namespace accessor methods (list, str, struct) that provide
+convenient access to PyArrow compute functions through the expression API.
+"""
+
+from typing import Any
+
+import pandas as pd
+import pyarrow as pa
+import pytest
+
+import ray
+from ray.data.expressions import col
+
+
+def assert_df_equal(result: pd.DataFrame, expected: pd.DataFrame):
+    """Assert dataframes are equal, ignoring dtype differences."""
+    pd.testing.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
Here and elsewhere -- this functions makes assumptions about the output ordering, so the tests might fail unexpectedly if tasks finish out of order. Consider using `rows_same` instead

Reply:
> Also nothing here should be order dependent.

I don't think that's true.

```python
ds = ray.data.from_items([{"val": "hello"}, {"val": "world"}])
result = ds.with_column("rev", col("val").str.reverse()).to_pandas()
expected = pd.DataFrame({"val": ["hello", "world"], "rev": ["olleh", "dlrow"]})
assert_df_equal(result, expected)
```

For example, this dataset starts with two blocks and launches two tasks. If the second task finishes in an earlier scheduling loop than the first task, then the result will look like this:

```python
>>> result
     val    rev
0  world  dlrow
1  hello  olleh
```

I don't think it'll be likely, but many of our tests are flaky because of this exact sort of behaviour, so I'd prefer not to make assumptions

Reply:
Okay to merge as-is to avoid blocking you. I can make `rows_same` more robust as a follow up